combat/CombatDamage.cpp

```diff
@@ -0,0 +1,211 @@
+#include "CombatDamage.h"
+
+#include "../util/Logger.h"
+#include "../util/GameRules.h"
+#include "../universe/Condition.h"
+#include "../universe/Enums.h"
+#include "../universe/Fighter.h"
+#include "../universe/Ship.h"
+#include "../universe/ShipDesign.h"
+#include "../universe/ShipPart.h"
+#include "../universe/ValueRef.h"
+
+// This knows about combat bouts, fighter launches, combatTargets conditions,
+// damage calculation (shortrange, fighters, shields)
+
+namespace {
+    ScriptingContext CombatContextWithShipTarget(std::shared_ptr<const Ship> source, float shields = 0.0f) {
+        // FIXME find default enemy - i use the given design as that at least exists (but might be non-targetable because of stealth)
+        //ErrorLogger() << "DESIGN " << GetPredefinedShipDesignManager().GetDesignID("08a58b08-0929-496d-84fc-faa91424ca21");
+        //auto target = std::make_shared<Ship>(ALL_EMPIRES, GetPredefinedShipDesignManager().GetDesignID("SD_MARK_1"), "SP_HUMAN");
+        //auto target = std::make_shared<Fighter>();
+        // XXX DesignID valid?
+        auto target = std::make_shared<Ship>(ALL_EMPIRES, source->DesignID(), "SP_HUMAN");
```

Comment:
you cannot assume the `SP_HUMAN` species exists in C++ code.

Reply:
> you cannot assume the `SP_HUMAN` species exists in C++ code.

Hm I need a ship instances for evaluation. 

Damage estimation probably should not rely on target species (even if weapon combatTargets could rely on species). Else one would always need to specify target species in some way - e.g. in case of a weapon only hurting lithic species.

I could use the source's species if that sounds better.

Reply:
> Damage estimation probably should not rely on target species

I would avoid imposing such limits on conditions for targetting

> I could use the source's species if that sounds better.

if it makes sense, sure...

Reply:
> > Damage estimation probably should not rely on target species
> 
> I would avoid imposing such limits on conditions for targetting
> 

Yes, targeting is free to do whatever one can script up.

Different from that, damage estimation is often without context, so a weapon only targeting psionic crewed ships should still do non-zero damage when estimating fleet values.